### PR TITLE
feat(parse): add method to parse files

### LINKED ISSUE #33: Implement `parse` method to parse data in files
Ref: #6 

**Methods:**

* [ ] `parse(data: {version: ?string, blocks: ?object[], decl: ?object[]): BEMEntityParts[]`

The method should determine the format of the file content.

**BEMDECL 1.0**

```js
exports.blocks = [
    { name: 'block' }
];
```

or

```js
exports.version = '1.0';
exports.decl = [
    { name: 'block' }
];
```

**BEMDECL 2.0**

```js
exports.version = '2.0';
exports.decl = [
    { block: 'block' }
];
```

**BEMDECL next**

```js
exports.version = 'next';
exports.decl = [
    { block: 'block' }
];
```

The method should translate any format to [normalized](https://github.com/bem/bem-decl/issues/15) **BEMDECL 2.0**.

```js
var parse = require('bem-decl').parse;

// bemdecl-1.0.js
// exports.blocks = [
//     { name: 'block' }
// ];

parse(fs.readFileSync('bemdecl-1.0.js')); // [{ block: 'block' }]
```

### PR DESCRIPTION
**Pull request by zxqfox**
Fixes #33

### CODE REVIEW

File: lib/parse.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const normalize = require('./normalize');
+const normalize2 = require('./normalize-harmony');
+
+const normalizers = {
+    '1.0': normalize,
+    '2.0': normalize2,
```
Comment (blond) [inline lib/parse.js]: `normalize-harmony` is not **BEMDECL v2.0**.

**BEMDECL v2.0** now is not supported (#3).

I think need throw error: "The BEMDECL 2.0 format is not supported.".

**Comment by blond** [comment]
Need add tests `should normalize decl` for each BEMDECL version.

I think need use `sinon`.

**Comment by zxqfox** [comment]
Tests for normalize are out of scope of this PR.

`sinon` ok.

**Comment by blond** [comment]
> Tests for normalize are out of scope of this PR.  For what purpose?

Need check that `normalize` will be called with the right options.

**Comment by zxqfox** [comment]
`parse` calls normalize internally with just one argument: `decl: object[]`.

**Comment by blond** [comment]
> parse calls normalize and normalize-harmony internally with just one argument: decl: object[].

Then check that calls necessary method with `decl`.

**Comment by zxqfox** [comment]
It is overkill, actually.

File: lib/parse.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+const normalizers = {
+    '1.0': normalize,
+    next: normalizeHarmony
+};
+
+/**
+ * Parses BEMDECL file data
+ *
+ * @param {{version: string, decl: BEMEntityPart[]}|{blocks: BEMEntityPart[]}} bemdecl [description]
```
Comment (Yeti-or) [inline lib/parse.js]: tenorok же какой еще part?

**Comment by zxqfox** [comment]
🆙

**Comment by blond** [comment]
👍

**Pull request merged**